### PR TITLE
GEN-2432 - feat(checkout v2): added payment page

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

image

Based on [design](https://www.figma.com/design/p1TEzLLVqJB9gKDNTDVGU4/Hedvig.com?node-id=470-2211&t=yFgWz55AmO4Ug3ql-4):

image

[Figma file](https://www.figma.com/design/p1TEzLLVqJB9gKDNTDVGU4/Hedvig.com?node-id=472-12230&t=yFgWz55AmO4Ug3ql-4)

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4500?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4500** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4500?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4494** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4494?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4493** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4493?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4492** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4492?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @guilhermespopolin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/[locale]/new/checkout/[shopSessionId]/payment/trustly/CheckoutPaymentTrustlyPage.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { useRouter } from 'next/navigation'
+import { useTranslation } from 'next-i18next'
+import { useState } from 'react'
+import { Text, yStack, theme } from 'ui'
+import { CheckIcon } from '@/components/ComparisonTable/ComparisonTable'
+import * as GridLayout from '@/components/GridLayout/GridLayout'
+import { TrustlyIframe } from '@/services/trustly/TrustlyIframe'
+import { successState, content } from './CheckoutPaymentTrustlyPage.css'
+
+const LOGGER = datadogLogs.createLogger('trustly')
```
Comment (alebedev) [inline apps/store/src/app/[locale]/new/checkout/[shopSessionId]/payment/trustly/CheckoutPaymentTrustlyPage.tsx:13]: `datadogLogger` ?  Feels like constant case should be only used for constants

File: apps/store/src/app/[locale]/new/checkout/[shopSessionId]/payment/trustly/CheckoutPaymentTrustlyPage.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      </div>
+    )
+  }
+
+  return (
+    <GridLayout.Root>
+      <GridLayout.Content className={content} width="1/3" align="center">
+        <div className={yStack({ gap: 'lg' })}>
+          <TrustlyIframe
+            style={{ width: 'min(29rem, 100%)', height: 'min(32rem, 60vh)' }}
```
Comment (alebedev) [inline apps/store/src/app/[locale]/new/checkout/[shopSessionId]/payment/trustly/CheckoutPaymentTrustlyPage.tsx:52]: should we extract this to CSS?

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/app/[locale]/new/checkout/[shopSessionId]/payment/trustly/CheckoutPaymentTrustlyPage.tsx (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/app/[locale]/new/checkout/[shopSessionId]/payment/trustly/CheckoutPaymentTrustlyPage.tsx:52]: In order for that to work I'd need to change `TrustlyIframe` so it uses vanilla extract. That will require a bunch of changes. This is the quickest solution but I can do it in the next PR.

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jul 22, 3:33 AM EDT**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4500).

**Pull request merged**